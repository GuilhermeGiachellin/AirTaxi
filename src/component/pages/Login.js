import {
  AnimatePresence, AnimateSharedLayout, motion, useCycle,
} from 'framer-motion';
import React from 'react';
import LogInForm from '../myForm/LogInForm';
import SignUpForm from '../myForm/SignUpForm';
import styles from './pages.module.scss';

const menuState = {
  intro: {
    goto: 'Login',
  },
  login: {
    goto: 'Signup',
  },
  signup: {
    goto: 'Back',
  },
};

const Login = () => {
  const [toggle, setToggle] = useCycle('intro', 'login', 'signup');
  return (
    <section
      className={styles.loginMainContainer}
      style={{
        display: 'flex', placeContent: 'center', flexDirection: 'column', width: '100%', alignItems: 'center',
      }}
    >
      <AnimateSharedLayout>
        <motion.div
          style={{
            display: 'flex', justifyContent: 'center', alignItems: 'center', flexDirection: 'column',
          }}
          layout
        >
          <motion.div layout>
            <h1> HELO HELO HELO I AM A TITLE </h1>
          </motion.div>
          <AnimatePresence exitBeforeEnter>
            { toggle === 'login' && (
            <motion.div
              key={1}
              animate={{ scale: 1 }}
              exit={{ scale: 0 }}
            >
              <LogInForm />
            </motion.div>
            )}
            { toggle === 'signup' && (
            <motion.div
              key={2}
              animate={{ scale: 1 }}
              exit={{ scale: 0 }}
            >
              <SignUpForm />
            </motion.div>
            )}
          </AnimatePresence>
        </motion.div>
        <motion.button className={styles.selectionBtn} type="button" onClick={() => setToggle()}>{menuState[toggle].goto}</motion.button>
      </AnimateSharedLayout>
    </section>
  );
};

export default Login;
