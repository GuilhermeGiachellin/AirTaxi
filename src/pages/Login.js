import {
  AnimatePresence, AnimateSharedLayout, motion, useCycle,
} from 'framer-motion';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import LoginInForm from '../component/myForm/LogInForm';
import SignUpForm from '../component/myForm/SignUpForm';
import PopUp from '../component/popups/PopUp';
import Layout from '../component/wrapper/Layout';
import { reset } from '../redux/slices/sessionSlice';
import styles from './pages.module.scss';

const menuState = {
  intro: {
    goto: 'Already have an account?',
  },
  login: {
    goto: 'Create account!',
  },
  signup: {
    goto: 'Back',
  },
};

const Login = () => {
  const [toggle, setToggle] = useCycle('intro', 'login', 'signup');
  const { status } = useSelector((state) => state.sessions);
  const dispatch = useDispatch();

  const onReset = () => {
    dispatch(reset());
  };

  return (
    <Layout preAnimation="fadeIn">
      <section
        className={styles.loginMainContainer}
        style={{
          display: 'flex', placeContent: 'center', flexDirection: 'column', width: '100vw', alignItems: 'center',
        }}
      >
        { status === 'error' && (
          <PopUp message="Wrong email or password!" handleInput={onReset} skip />
        )}
        <AnimateSharedLayout>
          <motion.div
            style={{
              display: 'flex', justifyContent: 'center', alignItems: 'center', flexDirection: 'column',
            }}
            layout
          >
            <motion.div layout>
              <h1 style={{ fontFamily: 'Righteous', color: 'white', fontSize: '3rem' }}> AIR TAXI </h1>
            </motion.div>
            <AnimatePresence exitBeforeEnter>
              { toggle === 'login' && (
              <motion.div
                key={1}
                initial={{ opacity: 0, x: '-200px' }}
                animate={{ opacity: 1, x: 0 }}
                exit={{ scale: 0, x: '200px' }}
              >
                <LoginInForm />
              </motion.div>
              )}
              { toggle === 'signup' && (
              <motion.div
                key={2}
                initial={{ opacity: 0, x: '-200px' }}
                animate={{ opacity: 1, x: 0 }}
                exit={{ scale: 0, x: '200px' }}
              >
                <SignUpForm />
              </motion.div>
              )}
            </AnimatePresence>
            <motion.button
              className={styles.selectionBtn}
              type="button"
              onClick={() => setToggle()}
            >
              {menuState[toggle].goto}
            </motion.button>
          </motion.div>

        </AnimateSharedLayout>
      </section>
    </Layout>
  );
};

export default Login;
