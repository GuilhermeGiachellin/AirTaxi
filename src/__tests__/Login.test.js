import React from 'react';
import {
  render, screen,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import '@testing-library/jest-dom/extend-expect';
import Login from '../pages/Login';

describe('Test Renders', () => {
  test('Render login title', async () => {
    render(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(screen.getByText(/AIR TAXI/i)).toBeInTheDocument();
  });
});

describe('Snapshoot of the Login page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(

      <Provider store={store}>
        <Login />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
