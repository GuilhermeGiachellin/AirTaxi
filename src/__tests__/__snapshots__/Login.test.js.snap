// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanpshoot of the Login page renders correctly 1`] = `
<main
  style={
    Object {
      "opacity": 0,
      "transform": "none",
    }
  }
>
  <section
    className="loginMainContainer"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "placeContent": "center",
        "width": "100vw",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={Object {}}
      >
        <h1
          style={
            Object {
              "color": "white",
              "fontFamily": "Righteous",
              "fontSize": "3rem",
            }
          }
        >
           AIR TAXI 
        </h1>
      </div>
      <button
        className="selectionBtn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Already have an account?
      </button>
    </div>
  </section>
</main>
`;
